### ISSUE TITLE
Merge the 3bot-v2 branch of tfchain into the master branch of tfchain

### ISSUE THREAD

**Issue by GlenDC** [threefoldtech/home#68]
(feel free to change to a better title, I couldn't come up with one)

- [x] Merge the 3bot-v2 branch of tfchain into the master branch of tfchain (8h) (@LeeSmet) (PR: https://github.com/threefoldfoundation/tfchain/pull/222)
It would include:

- all work already done for the improved 3bot transactions (with the biggest being the Rivine encoding, now part of Rivine (master));
- [x] one task that still has to be done, in order to make sure all our signatures of the different inputs and the Tx itself are unique, regardless if they come from different public keys or not (8h) (@glendc) (closed by https://github.com/threefoldfoundation/tfchain/commit/dd933fe89900302e7cbb27fcc2e3b726f49c38a8);
- a lot of work that currently only lives in the 3bot-v2 branch, because we can no longer merge master of rivine into master of tfchain, due to the fact that that would mean that we break the current 3bot transactions on testnet;

- ~Once that is done it would mean we need to reset testnet to the block before the first 3Bot Tx on testnet.~ (cancelled until we're ready for a testnet upgrade)

We would still need another reset though, as we still didn't allocate time in that case for including the 3Bot fee as a miner fee in the parent block. And even if we do this, we might still need a reset, because even if we fix the 3bot fee already, we would still not prevent double registration (of names/keys) in the same block, so should someone do so, a reset would still be required once that is preventable.

**Comment by GlenDC** [comment]
All that is left to do to finish this story, @robvanmieghem and @LeeSmet, is to do and check off:

> Once that is done it would mean we need to reset testnet to the block before the first 3Bot Tx on testnet.

Not sure if that was clear, or we need discussion about this first?
